### PR TITLE
feat(instance): add section to cover stream attach support

### PR DESCRIPTION
**Pull request by morphis**

### CODE REVIEW

File: howto/instance/create.md [hunk 1]
```diff
@@ -71,6 +71,22 @@ Launch a raw instance by providing the image ID in the following command:
 
 See [Provided images](https://discourse.ubuntu.com/t/provided-images/24185) for a list of images that are available in Anbox Cloud.
 
+### Launch an instance with streaming enabled
+
+*since 1.23.0*
+
+If you want to stream the visual output from an instance you have to specify this at creation time. If not being specified the instance cannot be streamed. In previous versions of Anbox Cloud it was only possible to create such an instance via the [Stream Gateway API](https://canonical.github.io/anbox-cloud.github.com/latest/anbox-stream-gateway/). The same can now be achieve through the [AMS API](https://canonical.github.io/anbox-cloud.github.com/latest/ams/) or the `amc` CLI.
```
Comment (ajanon) [inline howto/instance/create.md:78]:
```suggestion
If you want to stream the visual output from an instance you have to specify this at creation time. If not being specified the instance cannot be streamed. In previous versions of Anbox Cloud it was only possible to create such an instance via the [Stream Gateway API](https://canonical.github.io/anbox-cloud.github.com/latest/anbox-stream-gateway/). The same can now be achieved through the [AMS API](https://canonical.github.io/anbox-cloud.github.com/latest/ams/) or the `amc` CLI.
```

File: howto/instance/create.md [hunk 2]
```diff
@@ -71,6 +71,22 @@ Launch a raw instance by providing the image ID in the following command:
...
+
+To enable streaming for a new instance simply run
+
+    amc launch --enable-streaming ...
+
+AMS will automatically create a streaming session for the instance. You can find the id of the session as a tag on the instance in the format `session=<id>`.
+
+If you want to to further customize the display settings or frame rate you can use the `--display-size`, `--display-density` and `--fps` arguments. To create an instance with a 1080p resolution, a frame rate of 60 and a DPI of 120 you can run
```
Comment (ajanon) [inline howto/instance/create.md:86]:
```suggestion
If you want to further customise the display settings or frame rate you can use the `--display-size`, `--display-density` and `--fps` arguments. To create an instance with a 1080p resolution, a frame rate of 60 and a DPI of 120 you can run
```

**Review by ajanon — CHANGES_REQUESTED** [review]

File: howto/instance/create.md [hunk 3]
```diff
@@ -71,6 +71,22 @@ Launch a raw instance by providing the image ID in the following command:
 
 See [Provided images](https://discourse.ubuntu.com/t/provided-images/24185) for a list of images that are available in Anbox Cloud.
 
+### Launch an instance with streaming enabled
+
+*since 1.23.0*
+
+If you want to stream the visual output from an instance you have to specify this at creation time. If not being specified the instance cannot be streamed. In previous versions of Anbox Cloud it was only possible to create such an instance via the [Stream Gateway API](https://canonical.github.io/anbox-cloud.github.com/latest/anbox-stream-gateway/). The same can now be achieve through the [AMS API](https://canonical.github.io/anbox-cloud.github.com/latest/ams/) or the `amc` CLI.
+
+To enable streaming for a new instance simply run
```
Comment (keirthana) [inline howto/instance/create.md:80]:
```suggestion
To enable streaming for a new instance, simply run:
```

File: howto/instance/create.md (hunk 2 above)
Comment (keirthana) [inline howto/instance/create.md:86]: I feel this reads better, WDYT?

```suggestion
If you want to to further customise the streaming configuration such as display settings or frame rate, use the corresponding arguments: `--display-size`, `--display-density` and `--fps`. For example, to create an instance with a 1080p resolution, a frame rate of 60 and a DPI of 120, run:
```

**Review by keirthana — CHANGES_REQUESTED** [review]
Some minor comments

File: howto/instance/create.md [hunk 4]
```diff
@@ -71,6 +71,22 @@ Launch a raw instance by providing the image ID in the following command:
 
 See [Provided images](https://discourse.ubuntu.com/t/provided-images/24185) for a list of images that are available in Anbox Cloud.
 
+### Launch an instance with streaming enabled
+
+*since 1.23.0*
```
Comment (keirthana) [inline howto/instance/create.md:76]: The more we all start to contribute doc updates in advance, the more I feel like we should have separate branches for each release version. But that's a separate conversation. Let's discuss it further.

**Review by ajanon — APPROVED** [review]
LGTM

**Review by keirthana — APPROVED** [review]
LGTM!

File: howto/instance/create.md [hunk 5]
```diff
@@ -71,6 +71,22 @@ Launch a raw instance by providing the image ID in the following command:
...
+
+If you want to stream the visual output from an instance, specify it at the time of instance creation. Otherwise, the instance cannot be streamed. In previous versions of Anbox Cloud, it was only possible to create an instance that has streaming enabled via the [Stream Gateway API](https://canonical.github.io/anbox-cloud.github.com/latest/anbox-stream-gateway/). The same can now be achieved through the [AMS API](https://canonical.github.io/anbox-cloud.github.com/latest/ams/) or the `amc` CLI.
+
```
Comment (adglkh) [inline howto/instance/create.md:79]: Is it worth mentioning how the agent API token is created? Another question is who will be in charge of creating the agent API token? charm or end user?

Comment (morphis) [reply howto/instance/create.md:79]: It will all be automatically setup for the user in all cases (microbox, appliance or charms)

Comment (adglkh) [reply howto/instance/create.md:79]: Alright, that's the missing piece for now. 
We should create a story to track it.

Comment (adglkh) [reply howto/instance/create.md:79]: https://warthogs.atlassian.net/browse/AC-2415

**Review by adglkh — APPROVED** [review]
LGTM

File: reference/ams-configuration.md [hunk 6]
```diff
@@ -5,6 +5,9 @@ AMS provides various configuration items to customise its behaviour. The followi
 
 | Name | Type | Deprecated |    Default    | Description           |
 |-----|------|------------|---------------|-----------------------|
+| `agent.api.fingerprint` | string | No | - | Fingerprint of certificate in the AMS trust store which is trusted when communicating with the stream agent. |
```
Comment (keirthana) [inline reference/ams-configuration.md:8]: @morphis The changes need to go in `reference/ams-configuration.yaml`.

**Review by keirthana — CHANGES_REQUESTED** [review]

**Review by keirthana — APPROVED** [review]
LGTM

File: howto/instance/create.md (hunk 4 above)
Comment (keirthana) [inline howto/instance/create.md:76]: This should be 1.22.0

**Review by keirthana — CHANGES_REQUESTED** [review]
Missed the versioning in the previous review

File: howto/instance/create.md (hunk 4 above)
Comment (keirthana) [reply howto/instance/create.md]: Corrected it while resolving conflicts.

**Review by keirthana — APPROVED** [review]

**Pull request merged**